pyaedt/modeler/cad/Primitives.py

```diff
@@ -275,45 +587,4908 @@ def version(self):
+            coordinate system.
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+
+        References
+        ----------
+
+        >>> oEditor.SetWCS
+        """
+        if isinstance(name, BaseCoordinateSystem):
+            self.oeditor.SetWCS(
+                ["NAME:SetWCS Parameter", "Working Coordinate System:=", name.name, "RegionDepCSOk:=", False]
+            )
+        else:
+            self.oeditor.SetWCS(
+                ["NAME:SetWCS Parameter", "Working Coordinate System:=", name, "RegionDepCSOk:=", False]
+            )
+        return True
+
+    @pyaedt_function_handler()
+    def invert_cs(self, coordinate_system, to_global=False):
+        """Get the inverse translation and the conjugate quaternion of the input coordinate system.
+
+        By defining a new coordinate system with this information, the reference coordinate system
+        of the input coordinate system is obtained.
+
+        Parameters
+        ----------
+        coordinate_system : str, CoordinateSystem
+            Name of the destination reference system. A ``CoordinateSystem`` object can also be
+            used.
+        to_global : bool, optional
+            Whether to compute the inverse transformation of the input coordinate system with
+            respect to the global coordinate system. The default is ``False``.
+
+        Returns
+        -------
+        list
+            Origin coordinates.
+        list
+            Quaternion.
+        """
+        if isinstance(coordinate_system, BaseCoordinateSystem):
+            cs = coordinate_system
+        elif isinstance(coordinate_system, str):
+            cs_names = [i.name for i in self.coordinate_systems]
+            if coordinate_system not in cs_names:  # pragma: no cover
+                raise AttributeError("Specified coordinate system does not exist in the design.")
+            cs = self.coordinate_systems[cs_names.index(coordinate_system)]
+        else:  # pragma: no cover
+            raise AttributeError("coordinate_system must either be a string or a CoordinateSystem object.")
+
+        if to_global:
+            o, q = self.reference_cs_to_global(coordinate_system)
+            o = GeometryOperators.v_prod(-1, GeometryOperators.q_rotation(o, q))
+            q = [q[0], -q[1], -q[2], -q[3]]
+        else:
+            q = cs.quaternion
+            q = [q[0], -q[1], -q[2], -q[3]]
+            o = GeometryOperators.v_prod(-1, GeometryOperators.q_rotation(cs.origin, q))
+        return o, q
+
+    @pyaedt_function_handler()
+    def reference_cs_to_global(self, coordinate_system):
+        """Get the origin and quaternion defining the coordinate system in the global coordinates.
+
+        Parameters
+        ----------
+        coordinate_system : str, CoordinateSystem
+            Name of the destination reference system. The ``CoordinateSystem`` object can also be used.
```

Comment:
This is the other description that I was referring to earlier. It seems that the description for "name" and "coordinate_system" should be formatted similiarly.